solarkraft/src/verify.ts

```diff
@@ -185,6 +234,14 @@ export function verify(args: any) {
         'unverified' // only returned if array is empty
     )
 
+    const monitorStatus =
+        verificationStatus === 'ok'
+            ? MonitorAnalysisStatus.NoViolation
+            : MonitorAnalysisStatus.Violation
```

Comment:
Hmmm, so this would mark it as violation even if we have some internal error.

Looking at the code block above, I think it should reduce to `unverified` if there's an internal error on one of the monitors and no other monitor fails. Then we don't need any mapping from strings, but can reduce to a `MonitorAnalysisStatus` above directly.

Reply:
> you'd still want the alert contract to go off

But it will go off if you call it on each analysis, and if it changes from `NoViolation` to `Unknown` isn't that precisely what happens?

Reply:
> or migrate all 'ok' | 'fail' | 'unknown' to MonitorAnalysisStatus everywhere, which would also be an option

that's what I meant, but I haven't checked how big the impact would be

Reply:
Not too drastic, and it's probably cleaner to have one single way of enumerating outcomes, than 2 and convert between. I'll try to include this in this PR, unless I notice it getting extremely out-of-scope.